docs/retrieval-augmented-generation/rag-repo.md

```diff
@@ -0,0 +1,110 @@
+  - Con: If we put this out there as something that is intended to be useful in a variety of applications, the pressure is on us to make sure it is differentiated from other broadly applicable RAG capabilities.  Hopefully that will be true eventually, but it probably won't be true for a while.  It might make more sense to give this some time to mature as a local component of InstructLab before trying to spin it off as its own thing.
+  - Con: If we put it out there as its own open source project, that project needs all of the infrastructure of a full open source activity (governing structures, communication tools and protocols, etc.).  That's a lot of work to set up.  Keeping it inside InstructLab for now lets us keep using the infrastructure that InstructLab has for this purpose).
+  - Con: If we put it out there as its own open source project, it needs a name.  It is a lot of work to come up with a good name and there will be a lot of stakeholders with an interest in the name that comes up.
+- Keep the indexing and run-time RAG code in <https://github.com/redhat-et/PaRAGon> which is an emerging technologies prototype for this work.
+  - Mostly the same pros and cons as putting it in a new repo outside InstructLab plus the following:
+  - Pro: A prototype for the code we want is already there.
+  - Pro: It already has its own distinctive name (PaRAGon).
+  - Con: The existing repository has its own simple command-line interface which is useful for the prototype but we don't want it in the capability we release because too many command-line interfaces will confuse users.
+  - Con: The name PaRAGon seems fine to me, but probably more stakeholders need to weigh in on what a name would be.
+  - Con: The `redhat-et` label suggests that this is something "owned" by Red Hat which makes sense for the prototype but not so much for something we want a community to own in the long run.
+- Put the indexing and run-time RAG code in <https://github.com/instructlab/rag> AND keep the existing RAG functionality in that repository intact.
+  - Pro: It already exists.
+  - Pro: It avoids the confusion of having two different RAG repositories in <https://github.com/instructlab/>.
+  - Con: It creates the confusion of having two different RAG solutions in the same repository.  I guess we could mitigate that with developer documentation and marking legacy stuff as "deprecated".
+- Put the indexing and run-time RAG code in <https://github.com/instructlab/rag> AND eliminate the existing RAG functionality in that repository.
+  - Pro: It already exists.
+  - Pro: It avoids the confusion of having two different RAG repositories in <https://github.com/instructlab/>.
+  - Pro: It avoids the confusion of having two different RAG solutions since we'd be eliminating the old one.
+  - Con: That stuff must be there for a reason.  Maybe someone is using it.  It is not totally clear how we would even find out.  If someone can run this down maybe we can consider this option.
```

Comment:
I have to agree with the previous discussions that I think it makes sense to put everything in `instructlab/rag`. Hopefully, if any functionality is needed from what currently exists, we should be able to instrument a full-fledged RAG solution to be able to support that.

Also, more of a nit, I like the name `RAG` more for that repo. It keeps it in line with `SDG` and also just easier to say 😂

Reply:
@jwm4 have you considered renaming the existing repo, which has no dependencies on `instructlab` artifacts, to something like `rag-experiment` and instead use this name for the new `instructlab-rag` package?

Reply:
> @jwm4 have you considered renaming the existing repo, which has no dependencies on instructlab artifacts, to something like rag-experiment and instead use this name for the new instructlab-rag package?

That's being considered.  Another item being considered is moving the existing content in that repo into a branch and then putting the new stuff in main.  If we can get all the critical stakeholders on board then either of those would be fine with me.

Reply:
> Also, more of a nit, I like the name RAG more for that repo. It keeps it in line with SDG

I would also be fine with `RAG` instead of `rag` -- but definitely would not want _both_ in the game GH org.  So I am fine with renaming if that's the consensus of the reviewers.   If we get permission to use the existing repo, let's see if we can get a consensus around renaming it or not.

Reply:
>That's being considered. Another item being considered is moving the existing content in that repo into a branch and then putting the new stuff in main. If we can get all the critical stakeholders on board then either of those would be fine with me.

+1 to moving it to a branch and not introducing a new repo
we may want to communicate this sufficiently to the community (slack, email to dev@instructlab.ai etc) to give people a heads up before doing that 

alternatively we could also start developing the new approach in a branch before promoting it to main

Reply:
Right now it looks like this repository will not be concerned with the "G" part in "RAG" - namely, it's to provide retrieval functionality for use in InstructLab. Unless I'm missing something on that point, I advocate for sidestepping this problem and simply creating the new `instructlab/retrieve` repository to work in.

Reply:
No, the plan I was trying to communicate in this document was to include the generation too.  I will update to make this clearer.

Reply:
This con can be eliminated per Akash